PL_2_komentarze.py

```diff
@@ -0,0 +1,2722 @@
+from configparser import ConfigParser
+
+
+def patch_crypto_be_discovery():
+
+    """
+    Monkey patches cryptography's backend detection.
+    Objective: support pyinstaller freezing.
+    """
+
+    from cryptography.hazmat import backends
+
+    try:
+        from cryptography.hazmat.backends.commoncrypto.backend import backend as be_cc
+    except ImportError:
+        be_cc = None
+    try:
+        from cryptography.hazmat.backends.openssl.backend import backend as be_ossl
+    except ImportError:
+        be_ossl = None
+
+    backends._available_backends_list = [
+        be for be in (be_cc, be_ossl) if be is not None
+    ]
+
+patch_crypto_be_discovery()
+
+root = Tk()
+
+root.wm_minsize(923, 0)
+root.iconbitmap(r'bin\tubes.ico')
+
+versionnr = LooseVersion('0.5.9')
+
+Config = ConfigParser()
+
+def ConfigSectionMap(section):
+    Config.read('config.ini')
+    dict1 = {}
+    options = Config.options(section)
+    for option in options:
+        try:
+            dict1[option] = Config.get(section, option)
+        except:
+            print("exception on %s!" % option)
+            dict1[option] = None
+    return dict1
+
+def downloadupdate():
+    startfile('aktualizator.exe')
+    root.destroy()
+
+
+def checkupdate():
+    A = open('bin/adres.txt').read()
+    try:
+        x = open(A+r'/version.tub')
+        y = x.readline()
+        if 'version' in y:
+            _, z = y.split('=')
+        z1 = LooseVersion(z)
+        if z1 > versionnr:
+            p1 = messagebox.askyesno("Aktualizacja", "Znaleziono nową wersję programu.\nCzy chcesz ją teraz pobrać?", default="yes")
+            if p1:
+                downloadupdate()
+        else:
+            pass
+    except FileNotFoundError:
+        pass
+
+def wczytajoddzial():
+    try:
+        c = ConfigSectionMap('Uzytkownik')['oddzial']
+        ktoryoddzial.set(c)
+        d = ConfigSectionMap('Uzytkownik')['tooltip']
+        wlaczniktooltip.set(d)
+        e = ConfigSectionMap('Uzytkownik')['test']
+        wlaczniktest.set(e)
+    except:
+        ktoryoddzial.set('50')
+        Config.read('config.ini')
+        Config.add_section('Uzytkownik')
+        Config.set('Uzytkownik', 'oddzial', '50')
+        Config.set('Uzytkownik', 'tooltip', '0')
+        Config.set('Uzytkownik', 'test', '1')
+        x = open(r'config.ini', 'w')
+        Config.write(x)
+        messagebox.showwarning('Nie znaleziono konfiguracji oddziału!','Nie znaleziono konfiguracji oddziału.\nPrzywrócono domyślną konfigurację - PL50.\nAby ustawić inny oddział, skorzystaj z menu.')
+
+menubar = Menu(root)
+ktoryoddzial = StringVar()
+wlaczniktooltip = IntVar()
+wlaczniktest = IntVar()
+
+def setnumber():
+    try:
+        Config.read('config.ini')
+        Config.set('Uzytkownik', 'oddzial', str(ktoryoddzial.get()))
+        x = open('config.ini', 'w')
+        Config.write(x)
+        root.title("Świadectwomat v"+str(versionnr)+" - "+slownikoddzialow.get(ktoryoddzial.get()))
```

Comment:
Dobrze jest zostawić podstęp przed/po `+`, jest czytelniej. Przy sumowaniu większej liczby stringów wartu pokusić sie o poszukanie jakiejś funkcji, która zrobi to ładnie za nas. Przykład z dżawy `String.join()` podaje się elementy jako kolejne parametry.